rust-runtime/aws-smithy-http-server/src/protocols.rs

```diff
@@ -39,33 +39,75 @@ pub fn check_content_type<B>(
     }
 }
 
+pub fn accept_header_classifier<B>(req: &RequestParts<B>, content_type: &'static str) -> bool {
+    if req.headers().is_none() {
+        return true;
+    }
+    let headers = req.headers().unwrap();
+    if !headers.contains_key(http::header::ACCEPT) {
+        return true;
```

Comment:
Shouldn't we reject requests if there is no `Accept` header present?

Reply:
From [here](https://httpwg.org/specs/rfc7231.html#header.accept),

> A request without any Accept header field implies that the user agent will accept any media type in response.

We're complying with this behavior here (same in the other implementations of smithy)